pkg/auth/providers/azure/azure_actions.go

```diff
@@ -100,3 +106,47 @@ func (ap *azureProvider) testAndApply(actionName string, action *types.Action, r
 
 	return ap.tokenMGR.CreateTokenAndSetCookie(user.Name, userPrincipal, groupPrincipals, providerToken, 0, "Token via Azure Configuration", request, userExtraInfo)
 }
+
+// migrateToMicrosoftGraph represents the migration of the registered Azure AD auth provider
+// from the deprecated Azure AD Graph API to the Microsoft Graph API.
+// It modifies the existing auth config value in the database, so that it has a new endpoint to the new API.
+func (ap *azureProvider) migrateToMicrosoftGraph() error {
+	defer ap.deleteUserAccessTokens()
```

Comment:
I had thought about this, and I think it's fine. Deleting them does not hurt - users will just have to re-login.

Reply:
To Ricky's point here - I would expect that using the upgrade button would be a no-op on an upgraded provider. However, that's not really true. At minimum we are going to purge the group cache as well.

I don't think that this really represents an issue since both of the functions here are relatively benign, but this is something to think about.